server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecification.java

```diff
@@ -130,9 +130,11 @@ public class ModuleSpecification extends SimpleAttachable {
     private final List<PermissionFactory> permissionFactories = new ArrayList<>();
 
     public void addSystemDependency(final ModuleDependency dependency) {
-        if (!exclusions.contains(dependency.getIdentifier()) && !systemDependenciesSet.contains(dependency.getIdentifier())) {
-            this.systemDependencies.add(dependency);
-            this.systemDependenciesSet.add(dependency.getIdentifier());
+        if (!exclusions.contains(dependency.getIdentifier())) {
+            if (!systemDependenciesSet.contains(dependency.getIdentifier())) {
+                this.systemDependencies.add(dependency);
+                this.systemDependenciesSet.add(dependency.getIdentifier());
+            }
```

Comment:
just as a side note, I was expecting a different change here for this commit what WFCORE-6607 does is nothing regarding code change so not sure why we wanted it tracked in a different Jira. The Jira title suggests a change but there is no change at all, so, it could be quite confusing. I'm not asking for a change, just adding my observation.